examples/pytorch/rgcn/entity_sample_multi_gpu.py

```diff
@@ -0,0 +1,151 @@
+"""
+Differences compared to tkipf/relation-gcn
+* l2norm applied to all weights
+* remove nodes that won't be touched
+"""
+import argparse
+import gc
+import torch as th
+import torch.nn.functional as F
+import dgl.multiprocessing as mp
+import dgl
+
+from torchmetrics.functional import accuracy
+from torch.nn.parallel import DistributedDataParallel
+
+from entity_utils import load_data
+from entity_sample import init_dataloaders, init_models, train, evaluate
+
+def run(proc_id, n_gpus, n_cpus, args, devices, dataset, queue=None):
+    dev_id = devices[proc_id]
+    g, num_classes, num_rels, target_idx, inv_target, train_idx,\
+        test_idx, labels = dataset
+
+    dist_init_method = 'tcp://{master_ip}:{master_port}'.format(
+        master_ip='127.0.0.1', master_port='12345')
+    backend = 'nccl'
+
+    if not args.dgl_sparse:
+        backend = 'gloo'
+    if proc_id == 0:
+        print("backend using {}".format(backend))
+    th.distributed.init_process_group(backend=backend,
+                                      init_method=dist_init_method,
+                                      world_size=n_gpus,
+                                      rank=proc_id)
+
+    device = th.device(dev_id)
+    use_ddp = True if n_gpus > 1 else False
+    train_loader, val_loader, test_loader = init_dataloaders(
+        args, g, train_idx, test_idx, target_idx, dev_id, use_ddp=use_ddp)
+    embed_layer, model = init_models(args, device, g.num_nodes(), num_classes, num_rels)
+
+    labels = labels.to(device)
+    model = model.to(device)
+    model = DistributedDataParallel(model, device_ids=[dev_id], output_device=dev_id)
+    if not args.dgl_sparse:
+        embed_layer = DistributedDataParallel(embed_layer, device_ids=None, output_device=None)
```

Comment:
Generally speaking, I feel dgl_sparse introduces quite a lot complexity into the example code. Shall we remove it from the example code entirely? @classicsong What's your opinion?